python/pyspark/sql/types.py

```diff
@@ -57,7 +57,25 @@ def __ne__(self, other):
 
     @classmethod
     def typeName(cls):
-        return cls.__name__[:-4].lower()
+        typeTypeNameMap = {"DataType": "data",
+                           "NullType": "null",
+                           "StringType": "string",
+                           "BinaryType": "binary",
+                           "BooleanType": "boolean",
+                           "DateType": "date",
+                           "TimestampType": "timestamp",
+                           "DecimalType": "decimal",
+                           "DoubleType": "double",
+                           "FloatType": "float",
+                           "ByteType": "byte",
+                           "IntegerType": "integer",
+                           "LongType": "long",
+                           "ShortType": "short",
+                           "ArrayType": "array",
+                           "MapType": "map",
+                           "StructField": "struct",
```

Comment:
The reason I called  `typeName ` is, that I wanted to generate a HIVE table dynamically from data and to do this I need the type of each column. 

```
>>> sqlContext = SQLContext(sc)
>>> df = sqlContext.read.json('path_to_a_json_doc')
>>> cols = []
>>> for i in df.schema:
...   cols.append("`" + i.name + "`" + "\t" +  i.typeName())
>>> ",\n".join(cols)
```

In the real case, I converted the type name to a hive compatible form.